src/rgw/rgw_lc.cc

```diff
@@ -466,8 +466,10 @@ class LCObjsLister {
       if (ret < 0) {
         ldout(store->ctx(), 0) << "ERROR: list_op returned ret=" << ret << dendl;
         return ret;
-      } else {
-        obj_iter = objs.begin();
+      } 
+      obj_iter = objs.begin();
+      if (obj_iter == objs.end()) {
+        return false;
       }
       delay();
```

Comment:
yes, as I read this, the following code will execute after the delay():
   }
    *obj = *obj_iter;
which would have undefined behavior.

Matt